radix-engine-tests/tests/system/threading.rs

```diff
@@ -0,0 +1,268 @@
+use radix_engine_tests::common::PackageLoader;
+use scrypto_test::prelude::*;
+
+#[test]
+fn can_transfer_locked_bucket_between_threads() {
```

Comment:
I assume we have a test elsewhere which Josh added which checks we can't transfer a proof between threads? Maybe that should be moved here?

Reply:
I assumed so. Let me just add one.